src/main/java/se/sundsvall/parkingpermit/businesslogic/worker/AutomaticDenialDecisionTaskWorker.java

```diff
@@ -1,27 +1,94 @@
+
+import generated.se.sundsvall.casedata.ErrandDTO;
+import generated.se.sundsvall.casedata.StakeholderDTO;
+import se.sundsvall.parkingpermit.service.MessagingService;
 
 @Component
 @ExternalTaskSubscription("AutomaticDenialDecisionTask")
 public class AutomaticDenialDecisionTaskWorker extends AbstractWorker {
 	private static final Logger LOGGER = LoggerFactory.getLogger(AutomaticDenialDecisionTaskWorker.class);
+	private static final String PROCESS_ENGINE_FIRST_NAME = "Process";
+	private static final String PROCESS_ENGINE_LAST_NAME = "Engine";
+
+	@Autowired
+	private MessagingService messagingService;
 
 	@Override
 	public void execute(ExternalTask externalTask, ExternalTaskService externalTaskService) {
 		try {
-			LOGGER.info("Execute Worker for AutomaticDenialDecisionTask");
+			final var errand = getErrand(externalTask);
+			LOGGER.info("Executing automatic addition of dismissal to errand with id {}", errand.getId());
+
+			// PE needs to be added as stakeholder to the errand (if not already present) and store for later use when setting
+			// "decidedBy" on decision
+			final var stakeholder = ofNullable(errand.getStakeholders()).orElse(emptyList())
+				.stream()
+				.filter(AutomaticDenialDecisionTaskWorker::isProcessEngineStakeholder)
+				.findAny()
+				.orElseGet(() -> createProcessEngineStakeholder(errand));
+
+			final var props = messagingService.getProperties();
+			final var pdf = messagingService.renderPdf(errand);
+			final var decision = toDecision(FINAL, DISMISSAL, props.dismissalDescription())
+				.decidedBy(stakeholder)
+				.addLawItem(toLaw(props.lawHeading(), props.lawSfs(), props.lawChapter(), props.lawArticle()))
+				.addAttachmentsItem(toAttachment(BESLUT, props.filename(), "pdf", APPLICATION_PDF_VALUE, pdf));
+
+			caseDataClient.patchNewDecision(errand.getId(), decision);
 
 			externalTaskService.complete(externalTask);
 		} catch (Exception exception) {
-			LOGGER.error("Exception occurred in execution for task with id {} and businesskey {}", externalTask.getId(), externalTask.getBusinessKey());
+			LOGGER.error("Exception occurred in {} for task with id {} and businesskey {}", this.getClass().getSimpleName(), externalTask.getId(), externalTask.getBusinessKey(), exception);
 
 			failureHandler.handleException(externalTaskService, externalTask, exception.getMessage());
 		}
 	}
+
+	private StakeholderDTO createProcessEngineStakeholder(final ErrandDTO errand) {
+		final var id = extractStakeholderId(caseDataClient.addStakeholderToErrand(errand.getId(), toStakeholder(ADMINISTRATOR, PERSON, PROCESS_ENGINE_FIRST_NAME, PROCESS_ENGINE_LAST_NAME)));
+		return caseDataClient.getStakeholder(id);
+	}
+
+	private Long extractStakeholderId(final ResponseEntity<Void> response) {
+		return ofNullable(response.getHeaders().get(LOCATION)).orElse(emptyList()).stream()
+			.filter(Objects::nonNull)
+			.map(locationValue -> locationValue.substring(locationValue.lastIndexOf('/') + 1))
+			.map(Long::valueOf)
```

Comment:
Should we have a check before this row to ensure that the value is a Long-value (and not empty or a non-numeric string)?

Otherwise you will get a NumberFormatException, instead of a Problem with the text: "CaseData integration did not return any location for created stakeholder".